x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceService.java

```diff
@@ -70,31 +79,110 @@ public class ElasticInferenceService extends SenderService {
     public static final String NAME = "elastic";
     public static final String ELASTIC_INFERENCE_SERVICE_IDENTIFIER = "Elastic Inference Service";
 
-    private final ElasticInferenceServiceComponents elasticInferenceServiceComponents;
-
-    // The task types exposed via the _inference/_services API
-    private static final EnumSet<TaskType> SUPPORTED_TASK_TYPES_FOR_SERVICES_API = EnumSet.of(
-        TaskType.SPARSE_EMBEDDING,
-        TaskType.CHAT_COMPLETION
-    );
+    private static final Logger logger = LogManager.getLogger(ElasticInferenceService.class);
+    private static final EnumSet<TaskType> IMPLEMENTED_TASK_TYPES = EnumSet.of(TaskType.SPARSE_EMBEDDING, TaskType.CHAT_COMPLETION);
     private static final String SERVICE_NAME = "Elastic";
+    private static final ResponseHandler aclResponseHandler = createAclResponseHandler();
+
     /**
      * The task types that the {@link InferenceAction.Request} can accept.
      */
     private static final EnumSet<TaskType> SUPPORTED_INFERENCE_ACTION_TASK_TYPES = EnumSet.of(TaskType.SPARSE_EMBEDDING);
 
+    private final ElasticInferenceServiceComponents elasticInferenceServiceComponents;
+    private Configuration configuration;
+    private EnumSet<TaskType> enabledTaskTypes;
+    private final ModelRegistry modelRegistry;
+
     public ElasticInferenceService(
         HttpRequestSender.Factory factory,
         ServiceComponents serviceComponents,
-        ElasticInferenceServiceComponents elasticInferenceServiceComponents
+        ElasticInferenceServiceComponents elasticInferenceServiceComponents,
+        ModelRegistry modelRegistry
+    ) {
+        super(factory, serviceComponents);
+        this.elasticInferenceServiceComponents = Objects.requireNonNull(elasticInferenceServiceComponents);
+        this.modelRegistry = Objects.requireNonNull(modelRegistry);
+
+        enabledTaskTypes = EnumSet.noneOf(TaskType.class);
+        configuration = new Configuration(enabledTaskTypes);
+
+        init();
+
+        getAcl(elasticInferenceServiceComponents.elasticInferenceServiceUrl());
+    }
+
+    // TODO consider removing this should only be used for testing maybe mock getAcl instead?
+    ElasticInferenceService(
+        HttpRequestSender.Factory factory,
+        ServiceComponents serviceComponents,
+        ElasticInferenceServiceComponents elasticInferenceServiceComponents,
+        ModelRegistry modelRegistry,
+        ElasticInferenceServiceACL acl
     ) {
         super(factory, serviceComponents);
-        this.elasticInferenceServiceComponents = elasticInferenceServiceComponents;
+        this.elasticInferenceServiceComponents = Objects.requireNonNull(elasticInferenceServiceComponents);
+        this.modelRegistry = Objects.requireNonNull(modelRegistry);
+
+        setEnabledTaskTypes(acl);
+    }
+
+    private void getAcl(String baseEISUrl) {
```

Comment:
Nit: Let's make sure to call things just "authorizations" instead of "allowed models", and not mention "ACL"s. Just a bit less coupling between the authorization mechanism in EIS and what ES needs to know. EIS should be able to use authorization method, ACL or otherwise, and not confuse ES (i.e. not leak details ES doesn't need to know about).